src/CommunityToolkit/Xamarin.CommunityToolkit/Extensions/Navigation/NavigationExtensions.shared.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Threading.Tasks;
+using Xamarin.CommunityToolkit.UI.Views;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Extensions
+{
+	/// <summary>
+	/// Extension methods for <see cref="INavigation"/>.
+	/// </summary>
+	public static partial class NavigationExtensions
+	{
+		/// <summary>
+		/// Displays a popup.
+		/// </summary>
+		/// <param name="navigation">
+		/// The current <see cref="INavigation"/>.
+		/// </param>
+		/// <param name="popup">
+		/// The <see cref="BasePopup"/> to display.
+		/// </param>
+		public static void ShowPopup(this INavigation navigation, BasePopup popup)
+		{
+#if __ANDROID__
+			OnShowPopup(popup);
+#elif __IOS__
+			OnShowPopup(popup);
+#elif WINDOWS_UWP
+			OnShowPopup(popup);
```

Comment:
I'm not sure if I fully understand the Essentials approach. See code snippets below

Remove `#if` statements
**NavigationExtensions.shared.cs**
```c#
public static void ShowPopup(this INavigation navigation, BasePopup popup)
{
    OnShowPopup(popup);
}
```

**NavigationExtensions.netstandard.cs**
```c#
static void OnShowPopup(BasePopup popup)
{
    throw new NotSupportedException($"The current platform '{Device.RuntimePlatform}' does not support Xamarin Community Toolkit Popups.");
}
```

I would keep Android/iOS/UWP the same

Reply:
Yes, use the `FeatureNotSupportedException` https://github.com/xamarin/Essentials/blob/fe78dad8e2b0d78b7ba3ee5460862e1dbb2ed994/Xamarin.Essentials/Types/Shared/Exceptions.shared.cs#L33

Reply:
XCT doesn't have a direct dependency on Xamarin Essentials. Should I add the dependency.

I have updated this to follow the recommendations of @pictos but kept the `NotSupportedException` until we get an answer on the Xamarin Essentials issue

Reply:
@ahoefling I think that is best to grab that class from Essentials and bring it here. I can imagine that in the future we can use that more often.

Reply:
I needed to update the csproj file for .netstandard so expect to see that in the next code push as well
```
<Compile Include="**\*.netstandard.cs" />
<Compile Include="**\*.netstandard*.cs" />
```

Reply:
I started running into problems with this where it would generate duplicate files in the obj directory. I think this is because I don't have a setting right for netstandard 1.0 vs 2.0 vs 2.1. 

![image](https://user-images.githubusercontent.com/17751436/101970838-ca7d3c80-3bfa-11eb-97dc-0845b1d0ccb4.png)

Anyone have any ideas on this?

Reply:
I took another look at this tonight and figured out what was going on. I needed to make a few more changes to `csproj` and code than anticipated. I'll outline the changes here.

I added the following to the `csproj`
```xml
<PropertyGroup Condition=" $(TargetFramework.StartsWith('netstandard')) ">
  <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
  <GenerateTargetFrameworkAttribute>false</GenerateTargetFrameworkAttribute>
</PropertyGroup>
<ItemGroup Condition=" $(TargetFramework.StartsWith('netstandard')) ">
  <Compile Include="**/*.netstandard.cs" />
  <Compile Include="**/*.netstandard*.cs" />
</ItemGroup>
```

The trick to getting rid of the error in the screenshot above was in the new `PropertyGroup` section.

Once I had the `csproj` compiling correctly I could update the `NavigationExtensions` to follow the recommendations from Xamarin Essentials. The only catch was instead of just having a fallback class of `NavigationExtensions.netstandard.cs` I needed to create a very verbose file name of `NavigationExtensions.netstandard.macos.tvos.watchos.tizen.wpf.gtk.cs` which covers all of the platforms. If I did not add the platforms I would get a compile error as it won't be able to find the function `PlatformShowPopup`.